src/core/inc/types.h

```diff
@@ -36,6 +36,7 @@ typedef unsigned long cpumap_t;
 typedef unsigned irqid_t;
 
 typedef unsigned streamid_t;
+typedef unsigned deviceid_t;
```

Comment:
I'd say we should use the same type for arm and riscv. Maybe use `deviceid_t` for everything?

Reply:
Agree, `deviceid_t` is broader.